userspace/libsinsp/container_engine/cri.cpp

```diff
@@ -328,11 +300,11 @@ bool cri::resolve(sinsp_container_manager* manager, sinsp_threadinfo* tinfo, boo
 
 		if(!m_async_source)
 		{
-			auto async_source = new cri_async_source(manager, s_cri_timeout);
+			auto async_source = new cri_async_source(manager, m_cri.get(), s_cri_timeout);
 			m_async_source = std::unique_ptr<cri_async_source>(async_source);
```

Comment:
```c++
m_async_source = std::make_shared<cri_async_source>(manager, m_cri.get(), s_cri_timeout);
```

Reply:
Not sure I get the idea: do you want me to change unique_ptr to shared_ptr?